pkg/repositories/models/partition.go

```diff
@@ -0,0 +1,12 @@
+package models
+
+// Main Use cases:
+// 1. Filter artifacts by partition key/val in a dataset from UI [x]
+// 2. Get the artifact that has the partitions (x,y,z + tag_name) = latest [x]
```

Comment:
not sure if you're addressing this in the follow-up, but how does this model interplay with tags?

Reply:
It will make more sense in the followup PR with filters. Basically we will want to issue calls where we find all artifacts that have `tag=... AND partitionKey1=...`. That is why we put a